File: Benchmarks/Primes/IsPrime.php
<?php
function is_prime(int $n): bool
{
    if ($n < 2) {
        return false;
    }

    $i = 2;
    while ($i * $i <= $n) {
        if ($n % $i == 0) {
            return false;
        }
        $i++;
    }
    return true;
}

function count_primes(int $limit): int
{
    $count = 0;
    for ($i = 2; $i < $limit; $i++) {
        if (is_prime($i)) {
            $count++;
        }
    }
    return $count;
}

$limit = 100_000;
$count = count_primes($limit);

echo "Number of primes up to $limit: $count\n";
?>

File: Benchmarks/Primes/IsPrime.lua
local function is_prime(n)
    if n < 2 then
        return false
    end

    local i = 2
    while i * i <= n do
        if n % i == 0 then
            return false
        end
        i = i + 1
    end
    return true
end

local function count_primes(limit)
    local count = 0
    for i = 2, limit do
        if is_prime(i) then
            count = count + 1
        end
    end
    return count
end

local limit = 100000
local count = count_primes(limit)
print("Number of primes up to " .. limit .. ": " .. count)
